libraries/lib-files/FileNames.cpp

```diff
@@ -649,15 +649,16 @@ void FileNames::FindFilesInPathList(const wxString & pattern,
    }
 }
 
-bool FileNames::WritableLocationCheck(const FilePath& path)
+bool FileNames::WritableLocationCheck(const FilePath& path,
+                                    const wxString & message)
 {
     bool status = wxFileName::IsDirWritable(path);
 
     if (!status)
     {
         using namespace BasicUI;
         ShowMessageBox(
-            XO("Directory %s does not have write permissions").Format(path),
+            XO("%s as directory %s does not have write permissions").Format(message, path),
```

Comment:
To make the translations work with the grammars of all languages, it may be better that you just format two complete sentences.  Let the supplied messages all end in English with a full stop.   Compute `XO("%s does not have write permissions").Format(path)` and then use + to join them.